src/components/Common/Modal/CreateTrackModal/CreateTrackModal.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import { initialStateTrack } from '../../../../shared/initialStates';
+import { BUTTONS_TYPES, BUTTONS_NAMES, TRACK_FIELDS_KEYS } from '../../../../shared/constants';
+import { Button } from '../../../Buttons/Button/Button';
+import { ModalRow } from '../ModalRow/ModalRow';
+import style from './CreateTrackModal.module.css';
+import { createTrack, getTracks } from '../../../../services/tracks-services';
+import { generateId } from '../../../../shared/helpers';
+
+export class CreateTrackModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = initialStateTrack;
+  }
+
+  componentDidMount() {
+    const { userTasks, isEditMode, tracks, id } = this.props;
+    if (isEditMode) {
+      const trackData = tracks.filter((track) => track.id === id);
+      this.setState(trackData[0]);
+    } else {
+      const tasksNames = userTasks.map((task) => task.name) || [''];
+      this.setState({ name: tasksNames[0] });
```

Comment:
why do u need to take item by zero index ? rewrite it

Reply:
now the index is not needed. Used `find`
```suggestion
      this.setState(trackData);
```